lib/adapters/JasmineAdapter.js

```diff
@@ -4,68 +4,115 @@ import {Test, Suite} from "../Data.js";
     constructor(jasmine) {
         super();
+        this.jasmine = jasmine;
         jasmine.addReporter({
+            jasmineStarted: this.onJasmineStarted.bind(this),
             specDone: this.onSpecDone.bind(this),
             specStarted: this.onSpecStarted.bind(this),
             suiteStarted: this.onSuiteStarted.bind(this),
             suiteDone: this.onSuiteDone.bind(this),
             jasmineDone: this.onJasmineDone.bind(this)
         });
-        // a stack of test arrays. The top element on the stack is the currently active suite
-        this.tests = [[]];
-        // a stack of suite arrays
-        this.suites = [[]];
+
+        // Map internal jasmine ids to standardized test/suite objects.
+        this.idObjectMap = {};
     }
 
+    /**
+     * Recursively converts a jasmine suite into a standardized suite.
+     */
+    convertSuite(jasmineSuite) {
+        var name = jasmineSuite.description;
+        var childSuites = [];
+        var tests = [];
+        for (let child of jasmineSuite.children) {
+            // a child can be either a suite or a spec. We use the id to distinguish between the two.
+            if (child.id.indexOf("suite") === 0) {
+                childSuites.push(this.convertSuite(child));
+            } else if (child.id.indexOf("spec") === 0) {
+                tests.push(this.convertSpec(child));
+            }
+        }
+        var suite = new Suite(name, childSuites, tests);
+
+        this.idObjectMap[jasmineSuite.id] = suite;
 
-    onSpecStarted() {
-        this.startTime = new Date();
+        return suite;
     }
 
-    onSpecDone(details) {
-        var runtime = new Date() - this.startTime;
-        var testName = details.description;
-        var suiteName = details.fullName;
-        var errors = details.failedExpectations;
-        var status;
-        if (details.status === "pending") {
-            status = "skipped";
-        } else {
-            status = details.status;
-        }
+    /**
+     * Converts a jasmine spec into a standardized test.
+     */
+    convertSpec(jasmineSpec) {
+        var testName = jasmineSpec.description;
+        var suiteName = jasmineSpec.result.fullName;
 
         // Jasmine only provides a concatenated suiteName + testName string.
         // Remove the testName from the string.
         if (suiteName.lastIndexOf(testName) === suiteName.length - testName.length) {
             suiteName = suiteName.substr(0, suiteName.length - testName.length - 1);
         }
+
         var test = new Test(
             testName,
             suiteName,
-            status,
-            runtime,
-            errors
+            undefined,
+            undefined,
+            undefined
         );
+
+        this.idObjectMap[jasmineSpec.id] = test;
+
+        return test;
+    }
+
+
+    onJasmineStarted() {
+        var startSuite = this.convertSuite(this.jasmine.topSuite());
+        startSuite.name = "";
```

Comment:
This is similar to Mocha which has an empty string as the global suite title. Change anyway?